map boards display a map behind glass so just make them readable

db48x opened the pull request:
#### Summary
Features "allow the player to simply read map boards without forcing them to smash it"

#### Purpose of change
Fixes #84205

Currently you have to either pick the lock or smash the glass, then pick up the map, then read it. But the whole point of a map board is to display the map readably.

#### Describe the solution
I added an EOC to `f_map_board` that finds the map inside the board and activates it.

There currently can only be one examine action on a piece of furniture so I ditched the lockpicking. Map boards that have previously been picked will be migrated to smashed map boards. If you smash a map board then you must pick the map up out of the wreckage before you can read it.

#### Describe alternatives you've considered
I thought about occasionally making the map board dirty, such as by checking for bloodstains on the map tile, but that can be done later.

#### Testing
Wandered around until I found a trailhead. Read the map board.

db48x committed:
map boards display a map behind glass so just make them readable

Currently you have to either pick the lock or smash the glass, then
pick up the map, then read it. But the whole point of a map board is
to display the map readably.

I thought about occasionally making the map board dirty, such as by
checking for bloodstains on the map tile, but that can be done later.

There currently can only be one examine action on a piece of furniture
so I ditched the lockpicking. Map boards that have previously been
picked will be migrated to smashed map boards. If you smash a map
board then you must pick the map up out of the wreckage before you can
read it.

Fixes #84205

db48x committed:
reformat json just to add noise to the commit history

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

GuardianDll commented on data/json/furniture_and_terrain/furniture-storage.json:
```diff
@@ -1117,41 +1117,25 @@
+              "search_data": [ { "category": "maps" } ],
+              "title": "test",
+              "true_eocs": [ { "id": "EOC_ACTIVATE_MAP", "effect": [ { "u_activate": "reveal_map" } ] } ],
+              "false_eocs": [ { "id": "EOC_NO_MAP", "effect": [ { "u_message": "no map found" } ] } ]
```
```suggestion
```
This can be omitted completely, since it doesn't do anything

db48x commented:
Clearly I should have slept first before pushing this!

db48x committed:
clean up stuff left over from debugging

u_map_run_item_eocs already provides a message to the user in the case
of failure, although it’s not perfect for this situation. A second
message isn’t really useful.

db48x commented:
Ugh. Unrelated build failure.
